### ISSUE TITLE
Wrong vmagent query with dockerswarm_sd_configs filters

### ISSUE THREAD

**Issue by fannyfan414** [VictoriaMetrics/VictoriaMetrics#3579]
**Describe the bug**
I use docker swarm and try to migrate to VM from Prometheus (as full replacement) with **dockerswarm_sd_configs** [filters](https://docs.victoriametrics.com/sd_configs.html#dockerswarm_sd_configs), but vmagent stop working if i specify any filters. Without filters vmagent is working, but prometheus working with filters too.

**To Reproduce**
My docker-compose.yml file
```
version: '3.8'
  vmagent:
    image: victoriametrics/vmagent:v1.85.3
    expose:
      - 8429
    volumes:
      - monitoring_vmagent_data:/vmagentdata
      - /var/run/docker.sock:/var/run/docker.sock:ro
    configs:
      - source: monitoring_prometheus_config
        target: /etc/prometheus/prometheus.yml
    command:
      - "--promscrape.config=/etc/prometheus/prometheus.yml"
      - "--promscrape.config.strictParse=false"
      - "--remoteWrite.url=http://victoriametrics:8428/api/v1/write"
    networks:
      - monitoring_network
    restart: always
```

My prometheus.yml file
```
  - job_name: 'swarm_task_exporter'
    scrape_interval: 15s
    dockerswarm_sd_configs:
      - host: unix:///var/run/docker.sock
        role: tasks
        filters:
          - name: "desired-state"
            values: ["running"]
```

**Expected behavior**
If i connect to docker container and using curl create request, everything is ok
`curl --silent -XGET --unix-socket /run/docker.sock http://localhost/services?filters%5Bdesired-state%5D%5B0%5D=running`
But vmagent generate wrong request if i specify any filters

**Logs**
`error	VictoriaMetrics/lib/promscrape/config.go:1007	skipping dockerswarm_sd_config targets for job_name "swarm_task_exporter" because of error: cannot query dockerswarm api for services: unexpected status code returned from "http:///services?filters=%7B%22desired-state%22%3A%7B%22running%22%3Atrue%7D%7D": 400; expecting 200; response body: "{\"message\":\"Invalid filter 'desired-state'\"}\n`"

The logs show that there is a request for 

> http:///services?filters=%7B%22desired-state%22%3A%7B%22running%22%3Atrue%7D%7D

If you decode the parameters, you get 'filters' like this: `{"desired-state":{"running":true}}`
The correct parameters should be 'filters': `{"desired-state": "running"}`

**Version**
Swarm 20.10.22
vmagent v1.85.3

**Comment by valyala** [comment]
This issue has been fixed too in the [v1.79.7 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.7).

**Comment by fannyfan414** [comment]
@valyala @zekker6  Only partially fixed :)

Generation of query parameters now works correctly, but the request goes to the **wrong url**
In yml file i specify role: **tasks** 
```
  - job_name: 'swarm_task_exporter'
    scrape_interval: 15s
    dockerswarm_sd_configs:
      - host: unix:///var/run/docker.sock
        role: tasks
        filters:
          - name: "desired-state"
            values: ["running"]
```

so request should go to

>  unix/**tasks**?filters=%7B%22desired-state%22%3A%5B%22running%22%5D%7D

But now vmagent create request to 

> unix/**services**?filters=%7B%22desired-state%22%3A%5B%22running%22%5D%7D

 and return error because **services** don't really have such a filter, while **tasks** [do](https://docs.docker.com/engine/api/v1.41/#tag/Task/operation/TaskList)

>error	VictoriaMetrics/lib/promscrape/config.go:1039	skipping dockerswarm_sd_config targets for job_name "swarm_task_exporter" because of error: cannot query dockerswarm api for services: unexpected status code returned from "http://unix/services?filters=%7B%22desired-state%22%3A%5B%22running%22%5D%7D": 400; expecting 200; response body: "{\"message\":\"Invalid filter 'desired-state'\"}\n"

**Comment by zekker6** [comment]
Hi @fannyfan414, thank you for the report.

This issue was related to the label discovery for the tasks and have been fixed at https://github.com/VictoriaMetrics/VictoriaMetrics/pull/3632

In the meantime, you can build vmagent from sources by using instruction [here]( https://github.com/VictoriaMetrics/VictoriaMetrics#how-to-build-from-sources)
Or use pre-built docker image: `victoriametrics/vmagent:heads-fix-dockerswarm-discovery-filters-0-gf1c98e190`

**Comment by valyala** [comment]
The bugfix has been merged in the commit 8537533beb92a72c110b3c13e84fba2b25580637 . This commit will be included in the next release.

**Comment by fannyfan414** [comment]
Hi @zekker6 , thanks for fixing, in `version victoriametrics/vmagent:heads-fix-dockerswarm-discovery-filters-0-gf1c98e190` query is correct. But I found another bug or discrepancy with prometheus on the same topic.

I have 2 nodes(servers): redis + redis_exporter on each node. And third node for monitoring.

**Redis exporter** connected to 2 dockerswarm networks - **monitoring_network** and **redis_network**.
**Vmagent** connected to 1 network - **monitoring_network**

My docker-compose.yml
```
version: '3.8'
networks:
  redis_network:
    external:
      name: redis_network
  monitoring_network:
    external:
      name: monitoring_network

services:
  redis_exporter:
    image: oliver006/redis_exporter:v1.45.0
    expose:
      - 9121
#   Hostname important for prometheus metrics
    hostname: "{{.Service.Name}}.{{.Task.Slot}}"
    restart: unless-stopped
    networks:
      - redis_network
      - monitoring_network
    deploy:
      mode: replicated
      replicas: 2
        max_replicas_per_node: 1
        constraints:
          - "node.labels.node_group==redis"
      labels:
        - metrics_port=9121
        - metrics_path=/metrics
        
  vmagent:
    image: victoriametrics/vmagent:heads-fix-dockerswarm-discovery-filters-0-gf1c98e190
    expose:
      - 8429
    volumes:
      - monitoring_vmagent_data:/vmagentdata
      - /var/run/docker.sock:/var/run/docker.sock:ro
    configs:
      - source: monitoring_prometheus_config
        target: /etc/prometheus/prometheus.yml
    command:
      - "--promscrape.config=/etc/prometheus/prometheus.yml"
      - "--promscrape.config.strictParse=false"
      - "--remoteWrite.url=http://victoriametrics:8428/api/v1/write"
    networks:
      - monitoring_network
    restart: always
    deploy:
      mode: replicated
      replicas: 1
      placement:
        max_replicas_per_node: 1
        constraints:
          - "node.labels.node_group==monitoring"
```

Since the redis_exporter container is connected to **2 networks** (monitoring_network and redis_network), it has **two different** IP addresses for each network.

Docker gives Prometheus and vmagent through dockerwarm_sd_configs **all IPs from all networks**, but they can only access containers that are on the **monitoring_network**.

Therefore, I ask prometheus and vmagent to access metrics **not by IP**, but by **hostname** = docker service name + replica index number. It can be seen here **hostname** = {{.Service.Name}}.{{.Task.Slot}} and add labels to **metrics port and path**
```
services:
  redis_exporter:
    image: oliver006/redis_exporter:v1.45.0
    expose:
      - 9121
# Specify hostname
    hostname: "{{.Service.Name}}.{{.Task.Slot}}"
    deploy:
      ....
# Add labels for scrape config
      labels:
        - metrics_port=9121
        - metrics_path=/metrics
```

Therefore I use **relabel_configs** to change **__address__**
```
  - job_name: 'swarm_task_exporter'
    scrape_interval: 15s
    dockerswarm_sd_configs:
      - host: unix:///var/run/docker.sock
        role: tasks
        filters:
          - name: "desired-state"
            values: ["running"]
          - name: "service"
            values: [
              "redis_redis_exporter"
            ]
    relabel_configs:
      # Remove port from "__address__"
      - action: replace
        target_label: __address__
        source_labels:
          - __address__
        regex: ([^:]+)(?::\d+)?
        replacement: $1
      # Generate address from swarm container as "service_name.task_slot"
      - action: replace
        target_label: swarm_container_address
        source_labels:
          - __meta_dockerswarm_service_name
          - __meta_dockerswarm_task_slot
        separator: .
      # Remove task_slot=0 for global replicates containers
      - action: replace
        target_label: swarm_container_address
        source_labels:
          - swarm_container_address
          - __address__
        regex: (.*)\.[0];(.*)
        replacement: $2
      # Replace "__address__" as "swarm_container_address"
      - action: replace
        target_label: __address__
        source_labels:
          - swarm_container_address
          - __meta_dockerswarm_service_label_metrics_port
        separator: ":"
      # Replace metrics_path as label "metrics_path"
      - action: replace
        target_label: __metrics_path__
        source_labels:
          - __meta_dockerswarm_service_label_metrics_path
        regex: (.+)
```

If I open **prometheus UI** service-discovery I see **2 active targets** with __address__="redis_exporter.2:9121" and _address__="redis_redis_exporter.1:9121"

If open **vmagent UI** service-discovery, i see swarm_task_exporter (2/6 active) - 2 up and 4 dropped and **__address__** is **not changed** after relabelling: 
• active __address__="10.0.9.229:80" and __address__="10.0.19.32:80"
• dropped __address__="10.0.19.29:80", __address__="10.0.19.32:80", __address__="10.0.9.229:80", __address__="10.0.9.230:80"

And every scrape_interval i got error in vmagent log because **__address__ is not relabeled**:
> error	VictoriaMetrics/lib/promscrape/scraper.go:364	skipping duplicate scrape target with identical labels; endpoint=http://redis_exporter.1:9121/metrics, labels={instance="my",job="swarm_task_exporter",swarm_container_address="redis_exporter.1"}; make sure service discovery and relabeling is set up properly; see also https://docs.victoriametrics.com/vmagent.html#troubleshooting; original labels for target1: {__address__="10.0.9.230:80",__meta_dockerswarm_network_id="ule7l71hf291dtzqc7e0a4jfa",__meta_dockerswarm_network_ingress="false",__meta_dockerswarm_network_internal="false",__meta_dockerswarm_network_name="monitoring_network",__meta_dockerswarm_network_scope="swarm",__meta_dockerswarm_node_availability="active",__meta_dockerswarm_node_engine_version="20.10.22",__meta_dockerswarm_node_manager_leader="false",__meta_dockerswarm_node_manager_reachability="reachable",__meta_dockerswarm_node_platform_architecture="x86_64",__meta_dockerswarm_node_platform_os="linux",__meta_dockerswarm_node_role="manager",__meta_dockerswarm_node_status="ready",__meta_dockerswarm_service_id="xxah5tpto4kjbfour35c8piif",__meta_dockerswarm_service_label_com_docker_stack_image="oliver006/redis_exporter:v1.45.0",__meta_dockerswarm_service_label_metrics_path="/metrics",__meta_dockerswarm_service_label_metrics_port="9121",__meta_dockerswarm_service_mode="replicated",__meta_dockerswarm_service_name="redis_exporter",__meta_dockerswarm_service_task_container_hostname="{{.Service.Name}}.{{.Task.Slot}}",__meta_dockerswarm_service_task_container_image="oliver006/redis_exporter:v1.45.0@sha256:b3e847f29ae66fefcaf991e2ccee04b03e57314aab14abadfc87b0eed4453e65",__meta_dockerswarm_service_updating_status="completed",__meta_dockerswarm_task_container_id="a102c38f383d2331344ba9e1254174b75bc764a4c48f81d960b4d5d41aa7b502",__meta_dockerswarm_task_desired_state="running",__meta_dockerswarm_task_id="wf6d35lcmyr30bx9zxdbx4ppj",__meta_dockerswarm_task_slot="1",__meta_dockerswarm_task_state="running",__metrics_path__="/metrics",__scheme__="http",__scrape_interval__="15s",__scrape_timeout__="10s",job="swarm_task_exporter"}; original labels for target2: {__address__="10.0.19.32:80",__meta_dockerswarm_network_id="s8w9wn2j6t8qhfyvl5v8b1ls2",__meta_dockerswarm_network_ingress="false",__meta_dockerswarm_network_internal="false",__meta_dockerswarm_network_name="redis_network",__meta_dockerswarm_network_scope="swarm",__meta_dockerswarm_node_availability="active",__meta_dockerswarm_node_engine_version="20.10.22",__meta_dockerswarm_node_manager_leader="false",__meta_dockerswarm_node_manager_reachability="reachable",__meta_dockerswarm_node_platform_architecture="x86_64",__meta_dockerswarm_node_platform_os="linux",__meta_dockerswarm_node_role="manager",__meta_dockerswarm_node_status="ready",__meta_dockerswarm_service_id="xxah5tpto4kjbfour35c8piif",__meta_dockerswarm_service_label_com_docker_stack_image="oliver006/redis_exporter:v1.45.0",__meta_dockerswarm_service_label_metrics_path="/metrics",__meta_dockerswarm_service_label_metrics_port="9121",__meta_dockerswarm_service_mode="replicated",__meta_dockerswarm_service_name="redis_exporter",__meta_dockerswarm_service_task_container_hostname="{{.Service.Name}}.{{.Task.Slot}}",__meta_dockerswarm_service_task_container_image="oliver006/redis_exporter:v1.45.0@sha256:b3e847f29ae66fefcaf991e2ccee04b03e57314aab14abadfc87b0eed4453e65",__meta_dockerswarm_service_updating_status="completed",__meta_dockerswarm_task_container_id="a102c38f383d2331344ba9e1254174b75bc764a4c48f81d960b4d5d41aa7b502",__meta_dockerswarm_task_desired_state="running",__meta_dockerswarm_task_id="wf6d35lcmyr30bx9zxdbx4ppj",__meta_dockerswarm_task_slot="1",__meta_dockerswarm_task_state="running",__metrics_path__="/metrics",__scheme__="http",__scrape_interval__="15s",__scrape_timeout__="10s",job="swarm_task_exporter"}

**Comment by zekker6** [comment]
@fannyfan414 Could you check open vmagent UI by using `http://vmagent:8429/targets` and use `debug target relabeling` button next to the targets where relabeling rules were not applied?
This will display step-by-step evaluation of relabeling config which might help you find an issue.

**Comment by fannyfan414** [comment]
@zekker6 

All steps work out correctly, but from somewhere the last step appeared, which, as far as I understand, breaks everything:

`remove labels with __ prefix `

**Comment by zekker6** [comment]
@fannyfan414 Could you check if setting address to `instance` label instead of `__address__` helps?

**Comment by fannyfan414** [comment]
Maybe this auto-added step that removes all __ labels can be made optional via config?

Or it might be easier to remove everything started with __ except these **\_\_address\_\_** and **\_\_metrics_path\_\_** tags.
Or a little more difficult, keep track of the __ lables that have changed and don't delete them

**Comment by zekker6** [comment]
I've just tested your config with my docker-swarm installation and version from master, it does work for me just as expected:
Source labels:
```
{__address__="10.0.0.5:30000", __meta_dockerswarm_network_id="lmg1s8k7nv2ktuo59qbhxlubc", __meta_dockerswarm_network_ingress="true", __meta_dockerswarm_network_internal="false", __meta_dockerswarm_network_name="ingress", __meta_dockerswarm_network_scope="swarm", __meta_dockerswarm_node_address="10.12.2.15", __meta_dockerswarm_node_availability="active", __meta_dockerswarm_node_engine_version="20.10.22", __meta_dockerswarm_node_hostname="test-vmagent2", __meta_dockerswarm_node_id="r66n93h0n4s8kq47k9l2act4a", __meta_dockerswarm_node_manager_address="", __meta_dockerswarm_node_manager_leader="false", __meta_dockerswarm_node_manager_reachability="", __meta_dockerswarm_node_platform_architecture="x86_64", __meta_dockerswarm_node_platform_os="linux", __meta_dockerswarm_node_role="worker", __meta_dockerswarm_node_status="ready", __meta_dockerswarm_service_endpoint_port_name="", __meta_dockerswarm_service_endpoint_port_publish_mode="ingress", __meta_dockerswarm_service_id="3ggs9pjefwaeg31vzb5ztsxfn", __meta_dockerswarm_service_mode="replicated", __meta_dockerswarm_service_name="my_web", __meta_dockerswarm_service_task_container_hostname="", __meta_dockerswarm_service_task_container_image="nginx:latest@sha256:b8f2383a95879e1ae064940d9a200f67a6c79e710ed82ac42263397367e7cc4e", __meta_dockerswarm_service_updating_status="completed", __meta_dockerswarm_task_container_id="7df7e6cf7fd4f02a78fa0a355717b08ef9b9cff154cf000e960e8d93c486868d", __meta_dockerswarm_task_desired_state="running", __meta_dockerswarm_task_id="4sjrzt8gzdd32xvp85qq4az5k", __meta_dockerswarm_task_port_publish_mode="ingress", __meta_dockerswarm_task_slot="1", __meta_dockerswarm_task_state="running", __metrics_path__="/metrics", __scheme__="http", __scrape_interval__="1m0s", __scrape_timeout__="10s", job="dockerswarm"}
```

Resulted labels:
```
{instance="my_web.1:", job="dockerswarm", swarm_container_address="my_web.1"}
```

Labels starting with `__` are dropped after discovery as expected (same as Prometheus does). You can read [this](https://www.robustperception.io/life-of-a-label/) article which has a great explanation of labels lifecycle.

Could you share your resulting labels set after relabeling? Would be great to see relabel debug flow as well, to find a step where it does something wrong.

**Comment by fannyfan414** [comment]
@zekker6  Yes, you are right, I apologize for misleading, I just thought that the error was somehow related to the fact that it gives out in the vmagent logs. Prometheus also removes labels started with __

But why i got error every scrape_interval?
Prometheus works in parallel with the same config and does not give any errors.

`error VictoriaMetrics/lib/promscrape/scraper.go:364 skipping duplicate scrape target with identical labels; endpoint=http://redis_exporter.1:9121/metrics, labels={instance="my",job="swarm_task_exporter",swarm_container_address="redis_exporter.1"}; make sure service discovery and relabeling is set up properly; see also https://docs.victoriametrics.com/vmagent.html#troubleshooting; original labels for target1: {address="10.0.9.230:80",__meta_dockerswarm_network_id="ule7l71hf291dtzqc7e0a4jfa",__meta_dockerswarm_network_ingress="false",__meta_dockerswarm_network_internal="false",__meta_dockerswarm_network_name="monitoring_network",__meta_dockerswarm_network_scope="swarm",__meta_dockerswarm_node_availability="active",__meta_dockerswarm_node_engine_version="20.10.22",__meta_dockerswarm_node_manager_leader="false",__meta_dockerswarm_node_manager_reachability="reachable",__meta_dockerswarm_node_platform_architecture="x86_64",__meta_dockerswarm_node_platform_os="linux",__meta_dockerswarm_node_role="manager",__meta_dockerswarm_node_status="ready",__meta_dockerswarm_service_id="xxah5tpto4kjbfour35c8piif",__meta_dockerswarm_service_label_com_docker_stack_image="oliver006/redis_exporter:v1.45.0",__meta_dockerswarm_service_label_metrics_path="/metrics",__meta_dockerswarm_service_label_metrics_port="9121",__meta_dockerswarm_service_mode="replicated",__meta_dockerswarm_service_name="redis_exporter",__meta_dockerswarm_service_task_container_hostname="{{.Service.Name}}.{{.Task.Slot}}",__meta_dockerswarm_service_task_container_image="oliver006/redis_exporter:v1.45.0@sha256:b3e847f29ae66fefcaf991e2ccee04b03e57314aab14abadfc87b0eed4453e65",__meta_dockerswarm_service_updating_status="completed",__meta_dockerswarm_task_container_id="a102c38f383d2331344ba9e1254174b75bc764a4c48f81d960b4d5d41aa7b502",__meta_dockerswarm_task_desired_state="running",__meta_dockerswarm_task_id="wf6d35lcmyr30bx9zxdbx4ppj",__meta_dockerswarm_task_slot="1",__meta_dockerswarm_task_state="running",metrics_path="/metrics",scheme="http",scrape_interval="15s",scrape_timeout="10s",job="swarm_task_exporter"}; original labels for target2: {address="10.0.19.32:80",__meta_dockerswarm_network_id="s8w9wn2j6t8qhfyvl5v8b1ls2",__meta_dockerswarm_network_ingress="false",__meta_dockerswarm_network_internal="false",__meta_dockerswarm_network_name="redis_network",__meta_dockerswarm_network_scope="swarm",__meta_dockerswarm_node_availability="active",__meta_dockerswarm_node_engine_version="20.10.22",__meta_dockerswarm_node_manager_leader="false",__meta_dockerswarm_node_manager_reachability="reachable",__meta_dockerswarm_node_platform_architecture="x86_64",__meta_dockerswarm_node_platform_os="linux",__meta_dockerswarm_node_role="manager",__meta_dockerswarm_node_status="ready",__meta_dockerswarm_service_id="xxah5tpto4kjbfour35c8piif",__meta_dockerswarm_service_label_com_docker_stack_image="oliver006/redis_exporter:v1.45.0",__meta_dockerswarm_service_label_metrics_path="/metrics",__meta_dockerswarm_service_label_metrics_port="9121",__meta_dockerswarm_service_mode="replicated",__meta_dockerswarm_service_name="redis_exporter",__meta_dockerswarm_service_task_container_hostname="{{.Service.Name}}.{{.Task.Slot}}",__meta_dockerswarm_service_task_container_image="oliver006/redis_exporter:v1.45.0@sha256:b3e847f29ae66fefcaf991e2ccee04b03e57314aab14abadfc87b0eed4453e65",__meta_dockerswarm_service_updating_status="completed",__meta_dockerswarm_task_container_id="a102c38f383d2331344ba9e1254174b75bc764a4c48f81d960b4d5d41aa7b502",__meta_dockerswarm_task_desired_state="running",__meta_dockerswarm_task_id="wf6d35lcmyr30bx9zxdbx4ppj",__meta_dockerswarm_task_slot="1",__meta_dockerswarm_task_state="running",metrics_path="/metrics",scheme="http",scrape_interval="15s",scrape_timeout="10s",job="swarm_task_exporter"} `

**Comment by zekker6** [comment]
@fannyfan414 This container was added into discovery with 2 different networks. Since relabeling will produce two identical sets of labels vmagent will write this warning every time.
I can suggest you to keep one of targets since you only need to scrape targets from monitoring network:
```
- action: keep
  if: "{__meta_dockerswarm_network_name='monitoring_network'}"
```

**Comment by fannyfan414** [comment]
If i open `http://vmagent:8429/service-discovery` i saw:
• active` {__address__="10.0.9.230:80",  __meta_dockerswarm_network_name="monitoring_network", ...}` and `{__address__="10.0.9.229:80",  __meta_dockerswarm_network_name="monitoring_network", ...}` - my really 2 active container in **monitoring_network** 
• dropped `{__address__="10.0.19.29:80", __meta_dockerswarm_network_name="redis_network", ...}` and` {__address__="10.0.19.32:80", __meta_dockerswarm_network_name="redis_network", ...}` - container in **redis_network**, to which vmagent does not have access.

If I understand correctly, since they are marked as **dropped** , then vmagent **should not** query them.

But in the long log that I threw off, I see that there are two IPs: `10.0.9.230:80 `- active and `10.0.19.32:80` - dropped

**Comment by fannyfan414** [comment]
> @fannyfan414Этот контейнер был добавлен в обнаружение с двумя разными сетями. Поскольку перемаркировка создаст два идентичных набора меток, vmagent будет каждый раз выводить это предупреждение. Я могу предложить вам сохранить одну из целей, так как вам нужно только очищать цели из сети мониторинга:
> 
> ```
> - action: keep
>   if: "{__meta_dockerswarm_network_name='monitoring_network'}"
> ```

Thanks a lot, it helped!

**Comment by zekker6** [comment]
@fannyfan414 If both VM and Prometheus have similar workloads (ingest and query rates) that is not really a common case. Usually VM uses less resources than Prometheus when having similar load.
Could you check if your workload is similar between VM and Prometheus?
If it is similar, you could share [profiling](https://docs.victoriametrics.com/Single-server-VictoriaMetrics.html#profiling) stats so that we could check if there is a way we can optimize resources usage at VM side.

**Comment by zekker6** [comment]
@fannyfan414 Thanks for sharing profiling stats, as far as I can see memory usage is at expected level. 

Did you consider using discovery and scraping at vmsingle side?
vmsingle and vmagent share this functionality in-common, so when you're using both it will have larger memory footprint as you run 2 components.
In order to make this comparison kind of "fair" you could either run vmsingle with built-in scraping or to use vmagent remote-write for both vmsingle and prometheus.

**Comment by zekker6** [comment]
@fannyfan414 vmsingle is another name for single binary distribution of VictoriaMetric (according to configuration shared above you are using this one).


> Victoriametrics - stores metrics, vmagent - picks up and puts metrics in victoriametrics

This is one of options which we suggest by default as it has gives some better features than using single VictoriaMetrics. One of important features is that vmagent can buffer data in case VictoriaMetrics is not available due to some reason.

In case you need to minimize memory footprint you can configure VictoriaMetrics to scrape metrics just the same as vmagent.

**Comment by zekker6** [comment]
@fannyfan414 That might be related to the caches being reset after reboot. 
By default, VictoriaMetrics uses part of available memory for caching, so after time queries can populate cache to the previous level of RAM usage.
You can [tune caches](https://docs.victoriametrics.com/Single-server-VictoriaMetrics.html#cache-tuning) configuration, but usually this costs query execution speed since more data will be fetched from disk and use IO instead of RAM.

**Comment by valyala** [comment]
The original issue related to incorrect `filters` query arg passed to inpcompatible objects should be fixed in [v1.86.2](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing this issue as fixed then.

@fannyfan414 , could you open a new issue regarding memory/cpu usage in order to don't mix multiple issues into a single one? Just put a link to this issue in the newly created issue, so readers could get additional details.

**Comment by valyala** [comment]
FYI, this issue has been also fixed in [v1.79.8 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.8).
